src/tsolvers/egraph/EgraphSolver.cc

```diff
@@ -152,7 +152,6 @@ lbool Egraph::getPolaritySuggestion(PTRef p)
     if (term.size() > 2) { return l_Undef; } // For now focus on 2 arguments
     PTRef lhs = term[0];
     PTRef rhs = term[1];
-    assert(enode_store.has(lhs) && enode_store.has(rhs));
```

Comment:
Is there a reason to remove this assertion? Shouldn't we keep it?

Reply:
It fails.

Reply:
Wait, why does it suddenly fail?
It means we now do not create some enodes we were creating before?

Reply:
No, it doesn't look like that.  The failure happens in the regression test suite when SAT solver asks for a polarity suggestion.  I believe that it must have been some weird artefact of using inconsistent information.

Reply:
I am still not sure about this.
How can it happen that there is an equality (or disequality) in the Egraph such that the there is no Enode for one of the child terms?

Reply:
So, for the purpose of a documentation:
There is no change in what Enodes are created. This happens for equalities inside an uninterpreted function.
The change is that before such equalities were marked as `known` but not as `informed`.
The code is `Egraph::getPolaritySuggestion` checks `informed`. Thus previously the execution stopped at this check, but now it gets past it.

Reply:
Great, thanks!  I get the feeling that you agree that the new flow makes more sense?

It might be worth investigating if `isInformed` could be made virtual and the relevant information be obtained directly from the theory solver, possibly more efficiently.

Reply:
Can you please add a comment in the source code about when this condition can be violated?
I don't have a better solution now, so I think we can use your suggestion.

Reply:
Ok, I added a comment there and separated the assertion removal + comment to a distinct commit for maximal visibility.